analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -554,6 +556,35 @@ private String writeProjectNameIfNeeded() throws IOException {
         return newProjectName;
     }
 
+    private String extractProjectNameFromScmUrl(File rootDir) throws IOException {
+        File gitConfig = new File(new File(rootDir, ".git"), "config");
+        if (!gitConfig.isFile()) {
+            throw new ManipulationUncheckedException("No .git/config file found, failed to determine the root project name");
+        }
+        try {
+            List<String> lines = FileUtils.readLines(gitConfig, Charset.defaultCharset());
+            Optional<String> maybeScmUrlLine = lines.stream()
+                  .filter(line -> line.matches(SCM_URL_LINE_EXPR))
+                  .findFirst();
+
+            if (!maybeScmUrlLine.isPresent()) {
+                throw new ManipulationUncheckedException(
+                      ".git/config file doesn't define SCM URL, failed to determine the root project name");
+            }
+
+            // the line should be of form
+            // url = .../.../[^\\.]+(.git)?
+            String line = maybeScmUrlLine.get();
+            if (line.endsWith("/")) {
```

Comment:
When would it end with '/' ?

Reply:
hmm... it won't on Repour as it works now, I guess. But if you take a Github URL like https://github.com/quarkusio/quarkus/ you can easily clone the repo and get `url = https://github.com/quarkusio/quarkus/` in the `.git/config` (just verified that)